src/DoctrineMongoODMModule/Paginator/Adapter/DoctrinePaginator.php

```diff
@@ -19,7 +20,7 @@ class DoctrinePaginator implements AdapterInterface
     /**
      * Constructor
      */
-    public function __construct(Cursor $cursor)
+    public function __construct(Iterator $cursor)
```

Comment:
This change seems to widen the scope of acceptance and can potentially break code, as on line 34 if this file you rely on an undefined `Iterator` method called `getBaseCursor`.
When widen acceptance scope, it's recommended to add a test that covers the baseline and make sure it supports the baseline properly. In this case, a direct simple implementation of `Iterator`